bindgen/private/bindgen.bzl

```diff
@@ -220,6 +316,17 @@ rust_bindgen = rule(
             allow_single_file = True,
             mandatory = True,
         ),
+        "linkstatic": attr.bool(
```

Comment:
Yeah, I'm not sure about this naming, as it's not the same as `cc_library`'s `linkstatic`, but I don't have a better one off the top of my head... What may motivate someone to set this to `False`? Could you describe in a sentence what the `False` version of this attr does/means?

Reply:
if you have 
```
cc_lib -> bindgen -> lib_a -> lib_b -> bin
```
the `False` version of this flag will be the equivalent to the following when linking.
```
cc_lib -> bindgen -> lib_a -> lib_b -> bin
    \__________________________________/
```
This can cause link order issues that are annoying to deal with.

Reply:
I see... When would someone _want_ the other behavior?

Reply:
The `False` behavior? I couldn't tell you... My hope was only to leave the escape hatch for the situations I couldn't think of.

Reply:
Cool, let's ship this as is, treating `leak_symbols` as just a "way to get unblocked if this release breaks you" mechanism (and add a print warning if it's set asking people to file an issue), and aim to delete the setting completely in a few months?

Reply:
That's the plan!